File: FlickrWatchObjC/FlickrWatchObjC WatchKit Extension/SettingsInterfaceController.m
//
//  SettingsInterfaceController.m
//  FlickrWatchObjC
//
//  Created by Chad Etzel on 3/15/15.
//  Copyright (c) 2015 Charles Etzel. All rights reserved.
//

#import "SettingsInterfaceController.h"
#import "SettingsRowController.h"


@interface SettingsInterfaceController()

@property (nonatomic, weak) IBOutlet WKInterfaceTable *table;

@end


@implementation SettingsInterfaceController

- (void)awakeWithContext:(id)context
{
    [self.table setRowTypes:@[@"HourType1224Row"]];
    [self.table setNumberOfRows:2 withRowType:@"HourType1224Row"];
    //[self _reloadTableData];

    for (NSInteger i = 0; i < self.table.numberOfRows; i++) {
        if (i == 0) {
            SettingsRowController *row = [self.table rowControllerAtIndex:0];
            [row.label setText:@"12hr"];
            [row.checkImage setImageNamed:@"bluecheckcached"];
            if ([[NSUserDefaults standardUserDefaults] integerForKey:@"FlickrWatchHourModeKey"] == 0) {
                [row.checkImage setHidden:NO];
            } else {
                [row.checkImage setHidden:YES];
            }
        } else if (i == 1) {
            SettingsRowController *row = [self.table rowControllerAtIndex:1];
            [row.label setText:@"24hr"];
            [row.checkImage setImageNamed:@"bluecheckcached"];
            if ([[NSUserDefaults standardUserDefaults] integerForKey:@"FlickrWatchHourModeKey"] == 1) {
                [row.checkImage setHidden:NO];
            } else {
                [row.checkImage setHidden:YES];
            }
        }
    }
}

- (void)willActivate {
    // This method is called when watch view controller is about to be visible to user
    NSLog(@"%@ will activate", self);
}

- (void)didDeactivate {
    // This method is called when watch view controller is no longer visible
    NSLog(@"%@ did deactivate", self);
}

- (void)table:(WKInterfaceTable *)table didSelectRowAtIndex:(NSInteger)rowIndex
{
    NSLog(@"selected row!");

    NSInteger previousMode = [[NSUserDefaults standardUserDefaults] integerForKey:@"FlickrWatchHourModeKey"];

    [[NSUserDefaults standardUserDefaults] setObject:@(rowIndex) forKey:@"FlickrWatchHourModeKey"];
    [[NSUserDefaults standardUserDefaults] synchronize];

    if (previousMode != rowIndex) {
        [self _reloadTableData];
    }

}

- (void)_reloadTableData
{
    for (NSInteger i = 0; i < self.table.numberOfRows; i++) {
        if (i == 0) {
            SettingsRowController *row = [self.table rowControllerAtIndex:0];
            if ([[NSUserDefaults standardUserDefaults] integerForKey:@"FlickrWatchHourModeKey"] == 0) {
                [row.checkImage setHidden:NO];
            } else {
                [row.checkImage setHidden:YES];
            }
        } else if (i == 1) {
            SettingsRowController *row = [self.table rowControllerAtIndex:1];
            if ([[NSUserDefaults standardUserDefaults] integerForKey:@"FlickrWatchHourModeKey"] == 1) {
                [row.checkImage setHidden:NO];
            } else {
                [row.checkImage setHidden:YES];
            }
        }
    }
}


@end

File: FlickrWatchSwift/FlickrWatchSwift WatchKit Extension/SettingsInterfaceController.swift
//
//  SettingsInterfaceController.swift
//  FlickrWatchSwift
//
//  Created by Chad Etzel on 3/15/15.
//  Copyright (c) 2015 Charles Etzel. All rights reserved.
//

import WatchKit
import Foundation


class SettingsInterfaceController: WKInterfaceController {

    @IBOutlet weak var settingsTable: WKInterfaceTable!

    override func awakeWithContext(context: AnyObject!) {
        NSLog("settings did awake")

        settingsTable.setRowTypes(["HourType1224Row"])
        settingsTable.setNumberOfRows(2, withRowType: "HourType1224Row")

        for i in 0..<settingsTable.numberOfRows {
            if i == 0 {
                var row: SettingsRowController = settingsTable.rowControllerAtIndex(0) as! SettingsRowController
                row.label.setText("12hr")
                row.checkImage.setImageNamed("bluecheckcached");
                if NSUserDefaults.standardUserDefaults().integerForKey("FlickrWatchHourModeKey") == 0 {
                    row.checkImage.setHidden(false)
                } else {
                    row.checkImage.setHidden(true)
                }

            } else if i == 1 {
                var row: SettingsRowController = settingsTable.rowControllerAtIndex(1)as! SettingsRowController
                row.label.setText("24hr")
                row.checkImage.setImageNamed("bluecheckcached")
                if NSUserDefaults.standardUserDefaults().integerForKey("FlickrWatchHourModeKey") == 1 {
                    row.checkImage.setHidden(false)
                } else {
                    row.checkImage.setHidden(true)
                }
            }
        }

    }

    override func willActivate() {
        // This method is called when watch view controller is about to be visible to user
        super.willActivate()
        NSLog("%@ will activate", self)
    }

    override func didDeactivate() {
        // This method is called when watch view controller is no longer visible
        NSLog("%@ did deactivate", self)
        super.didDeactivate()
    }

    override func table(table: WKInterfaceTable, didSelectRowAtIndex rowIndex: Int) {
        NSLog("selected row!")

        var previousMode = NSUserDefaults.standardUserDefaults().integerForKey("FlickrWatchHourModeKey")

        NSUserDefaults.standardUserDefaults().setInteger(rowIndex, forKey: "FlickrWatchHourModeKey")
        NSUserDefaults.standardUserDefaults().synchronize()

        if previousMode != rowIndex {
            _reloadTableData()
        }
    }

    func _reloadTableData() {
        for i in 0..<settingsTable.numberOfRows {
            if i == 0 {
                var row: SettingsRowController = settingsTable.rowControllerAtIndex(0) as! SettingsRowController
                if NSUserDefaults.standardUserDefaults().integerForKey("FlickrWatchHourModeKey") == 0 {
                    row.checkImage.setHidden(false)
                } else {
                    row.checkImage.setHidden(true)
                }

            } else if i == 1 {
                var row: SettingsRowController = settingsTable.rowControllerAtIndex(1)as! SettingsRowController
                if NSUserDefaults.standardUserDefaults().integerForKey("FlickrWatchHourModeKey") == 1 {
                    row.checkImage.setHidden(false)
                } else {
                    row.checkImage.setHidden(true)
                }
            }
        }
    }

}
